connectivity/FEATURE_BLE/source/generic/GapImpl.cpp

```diff
@@ -1182,17 +1197,17 @@ void Gap::on_scan_stopped(bool success)
     }
 #endif // BLE_FEATURE_EXTENDED_ADVERTISING
 
-    if (_scan_address_refresh && !wait_for_advertising_stop) {
-#if BLE_ROLE_BROADCASTER
+    if (!wait_for_advertising_stop) {
         if (restart_advertising) {
             _address_refresh_sets.clear(LEGACY_ADVERTISING_HANDLE);
             startAdvertising(LEGACY_ADVERTISING_HANDLE);
             _adv_started_from_refresh.set(LEGACY_ADVERTISING_HANDLE);
         }
+    }
 #endif // BLE_ROLE_BROADCASTER
 
-        _scan_address_refresh = false;
-        startScan();
+    if (_scan_requested) {
```

Comment:
should I postpone this if (wait_for_advertising_stop == true) so that the address can be refreshed?

Reply:
We discussed this offline and yes, we need that so I added it